### PR TITLE
Add Volunteer Hour Tracking

### PR DESCRIPTION
**Pull request by laveur**
Added support for tracking volunteer hours, including a dashboard that shows idle and active volunteers.

### CODE REVIEW

File: console/views_new/volunteers.py [hunk 1]
```diff
@@ -108,4 +117,95 @@ def get_redirect_url(self, *args, **kwargs):
...
+
+@method_decorator(decorators, name="dispatch")
+class VolunteerDashboardPageView(PageView):
+    template_name = 'console-volunteers-dashboard.html'
+
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data(**kwargs)
+        event = get_current_event()
+        active_tasks = VolunteerTask.objects.filter(task_end=None)
+        idle_volunteers = Volunteer.objects.filter(event=event).filter(volunteer_state=ApplicationState.STATE_ACCEPTED).exclude(id__in=VolunteerTask.objects.filter(task_end=None).values_list('volunteer', flat=True))
```
Comment (dunkyl) [inline console/views_new/volunteers.py:130]: Just double checking, a task end isn't supposed to be set in the future? That seems normal, but as-is someone could have their task end set up in advance and be listed as idle during their shift.

File: console/views_new/volunteers.py [hunk 2]
```diff
@@ -64,8 +69,12 @@ class VolunteerDetailsPageView(PageView):
     def get_context_data(self, **kwargs):
         context = super().get_context_data(**kwargs)
         volunteer = get_object_or_404(Volunteer, pk=kwargs['volunteer_id'])
-
+        tasks = VolunteerTask.objects.filter(volunteer=volunteer).annotate(task_hours=F('task_end') - F('task_start')).annotate(effective_hours=ExpressionWrapper(F('task_hours') * F('task_multiplier'), output_field=DurationField()))
+        total_hours = asyncio.run(VolunteerTask.objects.filter(volunteer=volunteer).annotate(task_hours=F('task_end') - F('task_start')).annotate(effective_hours=ExpressionWrapper(F('task_hours') * F('task_multiplier'), output_field=DurationField())).aaggregate(total_hours=Sum(F('effective_hours'))))
```
Comment (dunkyl) [inline console/views_new/volunteers.py:73]: `aggregate` is the sync version of `aaggregate`

File: console/templatetags/console_extras.py [hunk 3]
```diff
@@ -0,0 +1,30 @@
+from django import template
+from datetime import timedelta
+
+register = template.Library()
+
+@register.filter()
+def pretty_delta(timedeltaobj):
+    """Convert a datetime.timedelta object into Days, Hours, Minutes, Seconds."""
+    if not isinstance(timedeltaobj, timedelta) or timedeltaobj == None: return ""
```
Comment (dunkyl) [inline console/templatetags/console_extras.py:9]: I would raise, which would catch misuses in tests. For the None case like with `task_end`, I would either add a different branch here for specifically the None case, or combine it with `|default_if_none:""` in `console-volunteers-detail.html`

**Review by dunkyl — COMMENTED** [review]
I was able to get all of the functionality working in a manual experiment, and I did not encounter any errors or unexpected behavior. After this is merged, I can include new tests for it in #102.

**Review by dunkyl — APPROVED** [review]

File: console/views_new/volunteers.py (hunk 2 above)
Comment (laveur) [reply console/views_new/volunteers.py:73]: That would explain a lot >.> I didn't realize there was two different versions.

File: console/views_new/volunteers.py (hunk 1 above)
Comment (laveur) [reply console/views_new/volunteers.py:130]: That would be the assumption. I've updated the form to check for this as a validation error.

File: console/templates/console-volunteers-detail.html [hunk 4]
```diff
@@ -1,97 +1,161 @@
...
+                    <td>{{task.task_notes}}</td>
+                    <td>{{task.task_start}}</td>
+                    <td>{{task.task_end|default_if_none:""}}</td>
+                    <td>
+                        {% if task.task_hours != None %}
+                        {{task.task_hours|pretty_delta}}
+                        {% endif %}
+                    </td>
+                    <td>
+                        {% if task.effective_hours != None %}
```
Comment (dunkyl) [inline console/templates/console-volunteers-detail.html:122]: Doubled up handling of None

**Review by dunkyl — APPROVED** [review]
One tiny piece of dead-code.

**Review by dunkyl — APPROVED** [review]

**Pull request merged**